snapshot/restore/restore.go

```diff
@@ -87,8 +89,14 @@ func Entry(ctx context.Context, rep repo.Repository, output Output, rootEntry fs
 		}
 	}
 
+	// Control the depth of a restore. Default (options.MaxDepth = 0) is to restore to full depth.
+	sd := options.MaxDepth
+	if sd > 0 {
+		sd = -sd - 1
```

Comment:
this is slightly weird logic, how about we simply default maxDepth to maxint32 or something?

Reply:
You're right. It was convoluted. I tried to make it better.

Reply:
Done in e206514